diff --git a/django/template/__init__.py b/django/template/__init__.py
index 8764bfa..ec42723 100644
--- a/django/template/__init__.py
+++ b/django/template/__init__.py
@@ -259,10 +259,13 @@ class Parser(object):
         self.filters = {}
         for lib in builtins:
             self.add_library(lib)
+        self.root_nodelist = None
 
     def parse(self, parse_until=None):
         if parse_until is None: parse_until = []
         nodelist = self.create_nodelist()
+        if self.root_nodelist is None:
+            self.root_nodelist = nodelist
         while self.tokens:
             token = self.next_token()
             if token.token_type == TOKEN_TEXT:
